bundle/src/main/java/com/adobe/acs/commons/wcm/properties/shared/impl/SharedComponentPropertiesReferenceProvider.java

```diff
@@ -0,0 +1,189 @@
+            if (StringUtils.isNotEmpty(componentResourceType)) {
+                componentResourceTypes.add(componentResourceType);
+            }
+            if (resource.hasChildren()) {
+                Iterator<Resource> children = resource.listChildren();
+                while (children.hasNext()) {
+                    collectComponentResourceTypes(children.next());
+                }
+            }
+        }
+    }
+
+
+    /**
+     * Iterates through the given collection of resourceTypes and finds all the components that has
+     * shared dialog, then adds the root page of the given resource to the list.
+     *
+     * @param resource {@link Resource}
+     * @return references
+     */
+    private List<Reference> findSharedComponents(Resource resource) {
+        List<Reference> references = new ArrayList<>();
+
+        ResourceResolver resourceResolver = resource.getResourceResolver();
+        for (String resourceType : componentResourceTypes) {
+
+            Resource componentResource = getComponentResource(resourceType, resourceResolver);
+            if (componentResource == null) {
+                continue;
+            }
+
+            if (!componentResource.getResourceType().equalsIgnoreCase("cq:Component")) {
+                continue;
+            }
+
+            Resource sharedDialog = componentResource.getResourceResolver()
+                .getResource(componentResource.getPath() + "/dialogshared");
+            Resource globalDialog = componentResource.getResourceResolver()
+                .getResource(componentResource.getPath() + "/dialogglobal");
+            if (sharedDialog == null && globalDialog == null) {
+                continue;
+            }
+
+            Page rootPage = pageRootProvider.getRootPage(resource);
+            if (rootPage != null) {
+                references.add(new Reference("sharedProperties", getReferenceName(rootPage),
+                    rootPage.getContentResource(),
+                    rootPage.getLastModified() != null ? rootPage.getLastModified()
+                        .getTimeInMillis() : -1L));
+            }
+        }
+        return references;
+    }
+
+    /**
+     * Returns the Component Resource by searching the configured resolver search paths ["/libs", "/apps"]
+     * @param resourceType {@link String}
+     * @param resourceResolver {@link ResourceResolver}
+     * @return Resource
+     */
+    private Resource getComponentResource(String resourceType, ResourceResolver resourceResolver) {
+        String[] resolverSearchPaths = resourceResolver.getSearchPath();
+
+        for (String searchPath : resolverSearchPaths) {
+            String tempResourceType = searchPath + resourceType;
+            if (resourceResolver.getResource(tempResourceType) != null) {
+                resourceType = searchPath + resourceType;
+                return resourceResolver.getResource(resourceType);
```

Comment:
```suggestion
            Resource resource = resourceResolver.getResource(searchPath + resourceType);
            if (resource != null) {
                return resource;
```

Reply:
Updated the code with the suggestions. Thank you for reviewing it.